pkg/client/client.go

```diff
@@ -102,21 +102,22 @@ func (n *NutanixClientHelper) buildManagementEndpoint(ctx context.Context, nutan
 	// Attempt to build a provider from the NutanixCluster object
 	providerForNutanixCluster, err := n.buildProviderFromNutanixCluster(nutanixCluster)
 	if err != nil {
-		return nil, fmt.Errorf("error building an environment provider from NutanixCluster: %w", err)
-	}
-	if providerForNutanixCluster != nil {
+		log.Info(fmt.Sprintf("error building an environment provider from NutanixCluster: %w", err))
+	} else if providerForNutanixCluster == nil {
+		err = fmt.Errorf("[WARNING] prismCentral attribute was not set on NutanixCluster %s in namespace %s. Defaulting to CAPX manager credentials", nutanixCluster.Name, nutanixCluster.Namespace)
+	} else if providerForNutanixCluster != nil {
 		providers = append(providers, providerForNutanixCluster)
-	} else {
-		log.Info(fmt.Sprintf("[WARNING] prismCentral attribute was not set on NutanixCluster %s in namespace %s. Defaulting to CAPX manager credentials", nutanixCluster.Name, nutanixCluster.Namespace))
 	}
 
-	// Fallback to building a provider using the global CAPX manager credentials
-	providerForLocalFile, err := n.buildProviderFromFile()
 	if err != nil {
```

Comment:
I don't think we want to fallback if the user tried to provide creds and it was invalid (ie. returned an err)
IMO the fallback should only happen if the creds are nil, maybe something like https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/compare/main...dkoshkin:cluster-api-provider-nutanix:dkoshkin/fix-client-always-reading-from-file?expand=1

Reply:
we dont want to return from first err != nil unless we try local file reading so in your fix. line 105.

Reply:
I concur with @dkoshkin  here. The fallback should only happen in a nil, nil return case. If there's an error it needs to be propagated and not silenced.